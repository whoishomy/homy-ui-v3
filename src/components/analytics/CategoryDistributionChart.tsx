'use client';

import { PieChart, Pie, Cell, ResponsiveContainer, Legend } from 'recharts';

const data = [
  { name: 'Exercise', value: 35 },
  { name: 'Nutrition', value: 25 },
  { name: 'Sleep', value: 20 },
  { name: 'Stress', value: 20 },
];

const COLORS = ['#0088FE', '#00C49F', '#FFBB28', '#FF8042'];

export function CategoryDistributionChart() {
  return (
    <div className="bg-white dark:bg-gray-800 rounded-lg p-6 shadow-sm">
      <h2 className="text-lg font-semibold mb-4">Category Distribution</h2>
      <div className="w-full h-[300px]">
        <ResponsiveContainer width="100%" height="100%">
          <PieChart>
            <Pie
              data={data}
              cx="50%"
              cy="50%"
              labelLine={false}
              outerRadius={80}
              fill="#8884d8"
              dataKey="value"
            >
              {data.map((entry, index) => (
                <Cell key={`cell-${index}`} fill={COLORS[index % COLORS.length]} />
              ))}
            </Pie>
            <Legend />
          </PieChart>
        </ResponsiveContainer>
      </div>
    </div>
  );
}
