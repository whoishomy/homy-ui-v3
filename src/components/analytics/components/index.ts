export * from './InsightCard';
export * from './InsightList';
export * from './GoalCompletionRadar';
export * from './HealthTrendCards';
export * from './CategoryFilter';
export * from './TimeRangeSelector';
export * from './InsightsSummary';
export * from './MonthlyAdherenceCalendar';
export { InsightCard } from './InsightCard'; 