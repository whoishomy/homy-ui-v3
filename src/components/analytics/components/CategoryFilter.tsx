import React from 'react';
import { HealthCategory } from '@/types/analytics';

interface CategoryFilterProps {
  selected: Array<keyof typeof HealthCategory>;
  onChange: (categories: Array<keyof typeof HealthCategory>) => void;
}

export const CategoryFilter: React.FC<CategoryFilterProps> = ({
  selected,
  onChange,
}) => {
  const handleToggle = (category: keyof typeof HealthCategory) => {
    if (selected.includes(category)) {
      onChange(selected.filter((c) => c !== category));
    } else {
      onChange([...selected, category]);
    }
  };

  return (
    <div className="flex flex-wrap gap-2">
      {Object.entries(HealthCategory).map(([key, label]) => (
        <button
          key={key}
          onClick={() => handleToggle(key as keyof typeof HealthCategory)}
          className={`inline-flex items-center px-3 py-1.5 rounded-full text-sm font-medium transition-colors ${
            selected.includes(key as keyof typeof HealthCategory)
              ? 'bg-green-100 text-green-800'
              : 'bg-gray-100 text-gray-700 hover:bg-gray-200'
          }`}
        >
          {label}
        </button>
      ))}
    </div>
  );
}; 