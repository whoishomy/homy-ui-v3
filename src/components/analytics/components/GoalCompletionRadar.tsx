import React from 'react';
import {
  Radar,
  RadarChart,
  PolarGrid,
  PolarAngleAxis,
  PolarRadiusAxis,
  ResponsiveContainer,
  Legend,
} from 'recharts';
import { HealthCategory } from '@/types/analytics';

interface GoalCompletionRadarProps {
  data: {
    category: keyof typeof HealthCategory;
    completion: number;
    target: number;
  }[];
}

export const GoalCompletionRadar: React.FC<GoalCompletionRadarProps> = ({
  data,
}) => {
  const formattedData = data.map((item) => ({
    category: HealthCategory[item.category],
    completion: item.completion,
    target: item.target,
  }));

  return (
    <div className="w-full h-[400px]">
      <ResponsiveContainer width="100%" height="100%">
        <RadarChart cx="50%" cy="50%" outerRadius="80%" data={formattedData}>
          <PolarGrid stroke="#E5E7EB" />
          <PolarAngleAxis
            dataKey="category"
            tick={{ fill: '#6B7280', fontSize: 12 }}
          />
          <PolarRadiusAxis
            angle={30}
            domain={[0, 100]}
            tick={{ fill: '#6B7280', fontSize: 10 }}
          />
          <Radar
            name="Tamamlama"
            dataKey="completion"
            stroke="#10B981"
            fill="#10B981"
            fillOpacity={0.6}
          />
          <Radar
            name="Hedef"
            dataKey="target"
            stroke="#6B7280"
            fill="#6B7280"
            fillOpacity={0.3}
          />
          <Legend
            wrapperStyle={{
              paddingTop: '2rem',
            }}
          />
        </RadarChart>
      </ResponsiveContainer>
    </div>
  );
}; 