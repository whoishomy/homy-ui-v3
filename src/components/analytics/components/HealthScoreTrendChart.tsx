import React from 'react';
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ResponsiveContainer,
} from 'recharts';
import { format } from 'date-fns';
import { tr } from 'date-fns/locale';
import { TrendData } from '@/types/analytics';

interface HealthScoreTrendChartProps {
  data: TrendData[];
}

export const HealthScoreTrendChart: React.FC<HealthScoreTrendChartProps> = ({
  data,
}) => {
  const formattedData = data.map((item) => ({
    ...item,
    date: format(new Date(item.date), 'dd MMM', { locale: tr }),
  }));

  return (
    <div className="w-full h-[300px]">
      <ResponsiveContainer width="100%" height="100%">
        <LineChart
          data={formattedData}
          margin={{
            top: 5,
            right: 30,
            left: 20,
            bottom: 5,
          }}
        >
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis
            dataKey="date"
            tick={{ fill: '#6B7280' }}
            tickLine={{ stroke: '#6B7280' }}
          />
          <YAxis
            tick={{ fill: '#6B7280' }}
            tickLine={{ stroke: '#6B7280' }}
            domain={[0, 100]}
          />
          <Tooltip
            contentStyle={{
              backgroundColor: '#ffffff',
              border: '1px solid #e5e7eb',
              borderRadius: '0.375rem',
            }}
            labelStyle={{ color: '#111827' }}
            itemStyle={{ color: '#6B7280' }}
          />
          <Legend
            wrapperStyle={{
              paddingTop: '1rem',
            }}
          />
          <Line
            type="monotone"
            dataKey="value"
            name="Sağlık Skoru"
            stroke="#10B981"
            strokeWidth={2}
            dot={{
              fill: '#10B981',
              stroke: '#ffffff',
              strokeWidth: 2,
              r: 4,
            }}
            activeDot={{
              fill: '#10B981',
              stroke: '#ffffff',
              strokeWidth: 2,
              r: 6,
            }}
          />
          {data[0]?.target && (
            <Line
              type="monotone"
              dataKey="target"
              name="Hedef"
              stroke="#6B7280"
              strokeDasharray="5 5"
              strokeWidth={2}
              dot={false}
            />
          )}
        </LineChart>
      </ResponsiveContainer>
    </div>
  );
}; 