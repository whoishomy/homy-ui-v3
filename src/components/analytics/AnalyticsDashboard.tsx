'use client';

import React, { useEffect, useState } from 'react';
import { motion } from 'framer-motion';
import { Activity } from 'lucide-react';

import { Card } from '@/components/ui/Card';
import { useAnalyticsStore } from '@/store';
import { AnalyticsFilters, HealthCategory, TimeRange } from '@/types/analytics';
import { HealthScoreTrendChart } from '@/components/analytics/charts/HealthScoreTrendChart';
import { HealthTrendCards } from '@/components/analytics/components/HealthTrendCards';
import { GoalCompletionRadar } from '@/components/analytics/components/GoalCompletionRadar';
import { MonthlyAdherenceCalendar } from '@/components/analytics/components/MonthlyAdherenceCalendar';
import { InsightsSummary } from '@/components/analytics/components/InsightsSummary';
import { TimeRangeSelector } from '@/components/analytics/components/TimeRangeSelector';
import { CategoryFilter } from '@/components/analytics/components/CategoryFilter';
import { CategoryDistributionChart } from './CategoryDistributionChart';
import { DailyWeeklyTabs } from './DailyWeeklyTabs';

interface AnalyticsDashboardProps {
  className?: string;
}

export const AnalyticsDashboard: React.FC<AnalyticsDashboardProps> = ({ className }) => {
  const [filters, setFilters] = useState<AnalyticsFilters>(() => {
    const now = new Date();
    const startDate = new Date(now);
    startDate.setDate(now.getDate() - 7); // Default to last 7 days for weekly view

    return {
      timeRange: 'weekly',
      startDate,
      endDate: now,
      categories: Object.keys(HealthCategory) as Array<keyof typeof HealthCategory>,
      onlyActive: true,
    };
  });

  const { summary, insights, loading, error, fetchSummary, fetchInsights } = useAnalyticsStore();

  useEffect(() => {
    fetchSummary(filters);
    fetchInsights(filters);
  }, [filters, fetchSummary, fetchInsights]);

  const handleTimeRangeChange = (range: TimeRange) => {
    setFilters((prev) => {
      const now = new Date();
      const startDate = new Date(now);

      switch (range) {
        case 'daily':
          startDate.setDate(now.getDate() - 1);
          break;
        case 'weekly':
          startDate.setDate(now.getDate() - 7);
          break;
        case 'monthly':
          startDate.setMonth(now.getMonth() - 1);
          break;
        case 'yearly':
          startDate.setFullYear(now.getFullYear() - 1);
          break;
      }

      return {
        ...prev,
        timeRange: range,
        startDate,
        endDate: now,
      };
    });
  };

  const handleCategoryFilter = (categories: Array<keyof typeof HealthCategory>) => {
    setFilters((prev) => ({ ...prev, categories }));
  };

  if (loading) {
    return (
      <div className="flex items-center justify-center min-h-screen">
        <motion.div
          animate={{ rotate: 360 }}
          transition={{ duration: 1, repeat: Infinity, ease: 'linear' }}
          className="rounded-full h-32 w-32 border-t-2 border-b-2 border-green-500"
        />
      </div>
    );
  }

  if (error) {
    return (
      <div className="bg-red-50 border-l-4 border-red-400 p-4 m-4">
        <div className="flex">
          <div className="flex-shrink-0">
            <svg className="h-5 w-5 text-red-400" viewBox="0 0 20 20" fill="currentColor">
              <path
                fillRule="evenodd"
                d="M10 18a8 8 0 100-16 8 8 0 000 16zM8.707 7.293a1 1 0 00-1.414 1.414L8.586 10l-1.293 1.293a1 1 0 101.414 1.414L10 11.414l1.293 1.293a1 1 0 001.414-1.414L11.414 10l1.293-1.293a1 1 0 00-1.414-1.414L10 8.586 8.707 7.293z"
                clipRule="evenodd"
              />
            </svg>
          </div>
          <div className="ml-3">
            <h3 className="text-sm font-medium text-red-800">Veri yüklenirken bir hata oluştu</h3>
            <p className="mt-2 text-sm text-red-700">{error}</p>
          </div>
        </div>
      </div>
    );
  }

  return (
    <motion.div
      initial={{ opacity: 0, y: 20 }}
      animate={{ opacity: 1, y: 0 }}
      transition={{ duration: 0.5 }}
      className="w-full space-y-6"
    >
      <div className="flex flex-col gap-4">
        <h1 className="text-2xl font-bold">Analytics Dashboard</h1>

        {/* Daily/Weekly Tabs */}
        <DailyWeeklyTabs />

        {/* Health Score Card */}
        <div className="bg-white dark:bg-gray-800 rounded-lg p-6 shadow-sm">
          <h2 className="text-lg font-semibold mb-4">Daily Health Score</h2>
          <div className="text-3xl font-bold text-green-600">87%</div>
        </div>

        {/* Charts Grid */}
        <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
          <CategoryDistributionChart />
          <HealthScoreTrendChart />
        </div>
      </div>
    </motion.div>
  );
};

AnalyticsDashboard.displayName = 'AnalyticsDashboard';
