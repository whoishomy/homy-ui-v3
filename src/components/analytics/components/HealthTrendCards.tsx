import React from 'react';
import { Activity, TrendingUp, TrendingDown } from 'lucide-react';
import { AnalyticsSummary } from '@/types/analytics';

interface HealthTrendCardsProps {
  summary: AnalyticsSummary | null;
}

export const HealthTrendCards: React.FC<HealthTrendCardsProps> = ({ summary }) => {
  if (!summary) return null;

  const overallTrend = summary.trends[summary.trends.length - 1]?.value || 0;
  const previousTrend = summary.trends[summary.trends.length - 2]?.value || 0;
  const trendDiff = overallTrend - previousTrend;
  const trendPercentage = ((trendDiff / previousTrend) * 100).toFixed(1);

  const cards = [
    {
      title: 'Genel Sağlık Skoru',
      value: `${Math.round(overallTrend)}%`,
      trend: {
        value: trendPercentage,
        direction: trendDiff >= 0 ? 'up' : 'down',
      },
      icon: Activity,
    },
    {
      title: 'Ortalama Uyum',
      value: `${Math.round(summary.adherence.overall)}%`,
      trend: {
        value: '0.0',
        direction: 'up',
      },
      icon: TrendingUp,
    },
    {
      title: 'Hedef Tamamlama',
      value: `${Math.round(
        summary.goalCompletion.reduce((acc, curr) => acc + curr.completion, 0) /
          summary.goalCompletion.length
      )}%`,
      trend: {
        value: '0.0',
        direction: 'up',
      },
      icon: TrendingDown,
    },
  ];

  return (
    <div className="grid grid-cols-1 md:grid-cols-3 gap-4">
      {cards.map((card) => {
        const Icon = card.icon;
        const isPositive = card.trend.direction === 'up';

        return (
          <div
            key={card.title}
            className="bg-white rounded-lg shadow p-6 flex items-start space-x-4"
          >
            <div
              className={`p-3 rounded-full ${
                isPositive ? 'bg-green-100' : 'bg-red-100'
              }`}
            >
              <Icon
                className={`w-6 h-6 ${
                  isPositive ? 'text-green-600' : 'text-red-600'
                }`}
              />
            </div>
            <div>
              <p className="text-sm font-medium text-gray-600">{card.title}</p>
              <p className="text-2xl font-semibold text-gray-900">{card.value}</p>
              <div className="mt-1 flex items-center">
                {card.trend.direction === 'up' ? (
                  <TrendingUp className="w-4 h-4 text-green-500" />
                ) : (
                  <TrendingDown className="w-4 h-4 text-red-500" />
                )}
                <span
                  className={`ml-1 text-sm font-medium ${
                    isPositive ? 'text-green-600' : 'text-red-600'
                  }`}
                >
                  {card.trend.value}%
                </span>
                <span className="ml-1 text-sm text-gray-500">
                  geçen haftaya göre
                </span>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
}; 