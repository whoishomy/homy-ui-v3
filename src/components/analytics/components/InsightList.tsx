import React, { useState } from 'react';
import { Download } from 'lucide-react';
import { HealthInsight, InsightCategory } from '@/types/analytics';
import { InsightCard } from './InsightCard';
import { downloadInsightsCSV } from '@/utils/exportUtils';
import { Button } from '@/components/ui/button';
import { getInsightCategoryOptions } from '@/utils/insightCategory';

interface InsightListProps {
  insights: HealthInsight[];
  className?: string;
}

export const InsightList: React.FC<InsightListProps> = ({ insights, className }) => {
  const [selectedCategory, setSelectedCategory] = useState<InsightCategory | 'all'>('all');
  const categoryOptions = getInsightCategoryOptions();

  const filteredInsights =
    selectedCategory === 'all'
      ? insights
      : insights.filter((insight) => insight.category === selectedCategory);

  const handleExport = async () => {
    try {
      await downloadInsightsCSV(insights);
    } catch (error) {
      // Error handling will be implemented later
      console.error('Failed to export insights:', error);
    }
  };

  const handleDismiss = (id: string) => {
    // Handle dismiss logic if needed
  };

  return (
    <div className={className}>
      <div className="flex justify-between items-center mb-4">
        <h2 className="text-xl font-semibold text-gray-900">İçgörüler</h2>
        <div className="flex items-center gap-4">
          <select
            value={selectedCategory}
            onChange={(e) => setSelectedCategory(e.target.value as InsightCategory | 'all')}
            className="rounded-md border border-gray-300 text-sm"
          >
            <option value="all">Tüm Kategoriler</option>
            {categoryOptions.map(({ value, label }) => (
              <option key={value} value={value}>
                {label}
              </option>
            ))}
          </select>
          <Button
            onClick={handleExport}
            variant="outline"
            size="sm"
            className="flex items-center gap-2"
          >
            <Download className="w-4 h-4" aria-hidden="true" />
            CSV İndir
          </Button>
        </div>
      </div>

      <div className="space-y-4">
        {filteredInsights.length === 0 ? (
          <div className="text-center py-8 text-gray-500">
            {selectedCategory === 'all'
              ? 'Henüz içgörü bulunmuyor'
              : 'Bu kategoride içgörü bulunmuyor'}
          </div>
        ) : (
          filteredInsights.map((insight) => (
            <InsightCard key={insight.id} insight={insight} onDismiss={handleDismiss} />
          ))
        )}
      </div>
    </div>
  );
};
