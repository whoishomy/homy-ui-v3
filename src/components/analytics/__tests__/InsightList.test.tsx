import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { vi } from 'vitest';
import { axe, toHaveNoViolations } from 'jest-axe';
import { InsightList } from '../components/InsightList';
import type { HealthInsight } from '@/types/analytics';
import { InsightCategory } from '@/types/analytics';
import * as exportUtils from '@/utils/exportUtils';
import { mockInsightsList, createMockInsight } from './test-helpers/mockInsights';

expect.extend(toHaveNoViolations);

// Mock i18next
vi.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => {
      const translations: Record<string, string> = {
        direction: 'ltr',
        'insight.metrics': 'İlgili Metrikler',
        'insight.dismiss': 'Kapat',
      };
      return translations[key] || key;
    },
  }),
}));

// Mock the exportUtils
vi.mock('@/utils/exportUtils', () => ({
  downloadInsightsCSV: vi.fn(),
  generateCSVFromInsights: vi.fn(),
}));

const renderInsightList = (props: { insights: HealthInsight[]; className?: string }) => {
  return render(<InsightList {...props} />);
};

describe('InsightList', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('Rendering', () => {
    it('renders list of insights', () => {
      renderInsightList({ insights: mockInsightsList });
      expect(screen.getAllByRole('article')).toHaveLength(mockInsightsList.length);
    });

    it('renders empty state message when no insights', () => {
      renderInsightList({ insights: [] });
      expect(screen.queryByRole('article')).not.toBeInTheDocument();
    });

    it('applies custom className when provided', () => {
      const className = 'custom-class';
      const { container } = renderInsightList({ insights: mockInsightsList, className });
      expect(container.firstChild).toHaveClass(className);
    });

    it('renders correct icon and color for each insight type', () => {
      renderInsightList({
        insights: [
          createMockInsight('1', 'success'),
          createMockInsight('2', 'warning'),
          createMockInsight('3', 'error'),
        ],
      });

      expect(screen.getByTestId('insight-icon-success')).toHaveClass('text-green-500');
      expect(screen.getByTestId('insight-icon-warning')).toHaveClass('text-yellow-500');
      expect(screen.getByTestId('insight-icon-error')).toHaveClass('text-red-500');
    });

    it('renders persona information when available', () => {
      renderInsightList({
        insights: [createMockInsight('1', 'success', InsightCategory.PHYSICAL, true)],
      });

      const article = screen.getByRole('article');
      expect(article).toHaveAttribute('data-persona-id', 'test-persona');
    });
  });

  describe('Export functionality', () => {
    it('renders export button', () => {
      renderInsightList({ insights: mockInsightsList });
      const exportButton = screen.getByRole('button', { name: 'CSV İndir' });
      expect(exportButton).toBeInTheDocument();
    });

    it('calls downloadInsightsCSV when export button clicked', async () => {
      const user = userEvent.setup();
      renderInsightList({ insights: mockInsightsList });
      const exportButton = screen.getByRole('button', { name: 'CSV İndir' });

      await user.click(exportButton);
      expect(exportUtils.downloadInsightsCSV).toHaveBeenCalledWith(mockInsightsList);
    });

    it('generates correct CSV content', () => {
      renderInsightList({ insights: mockInsightsList });
      expect(exportUtils.downloadInsightsCSV).not.toHaveBeenCalled();
    });
  });

  describe('Insight dismissal', () => {
    it('handles insight dismissal', async () => {
      const user = userEvent.setup();
      renderInsightList({ insights: mockInsightsList });
      const dismissButtons = screen.getAllByRole('button', { name: 'Kapat' });

      await user.click(dismissButtons[0]);
      // Add assertions for dismiss logic when implemented
    });

    it('maintains focus after dismissal', async () => {
      const user = userEvent.setup();
      renderInsightList({ insights: mockInsightsList });

      await user.tab(); // Focus first interactive element
      const exportButton = screen.getByRole('button', { name: 'CSV İndir' });
      expect(exportButton).toHaveFocus();
    });
  });

  describe('Accessibility', () => {
    it('has proper heading structure', () => {
      renderInsightList({ insights: mockInsightsList });
      expect(screen.getByRole('heading', { name: 'İçgörüler' })).toBeInTheDocument();
    });

    it('export button has proper attributes', () => {
      renderInsightList({ insights: mockInsightsList });
      const exportButton = screen.getByRole('button', { name: 'CSV İndir' });
      const downloadIcon = exportButton.querySelector('svg');
      expect(downloadIcon).toHaveAttribute('aria-hidden', 'true');
    });

    it('passes axe accessibility tests', async () => {
      const { container } = renderInsightList({ insights: mockInsightsList });
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });

    it('supports keyboard navigation between insights', async () => {
      const user = userEvent.setup();
      renderInsightList({ insights: mockInsightsList });

      await user.tab(); // Focus first interactive element
      const exportButton = screen.getByRole('button', { name: 'CSV İndir' });
      expect(exportButton).toHaveFocus();

      await user.tab(); // Focus first dismiss button
      const dismissButton = screen.getAllByRole('button', { name: 'Kapat' })[0];
      expect(dismissButton).toHaveFocus();
    });
  });

  describe('Error Handling', () => {
    it('handles export errors gracefully', async () => {
      const user = userEvent.setup();
      const mockDownload = vi.mocked(exportUtils.downloadInsightsCSV);
      mockDownload.mockImplementationOnce(() => Promise.reject(new Error('Export failed')));

      renderInsightList({ insights: mockInsightsList });
      const exportButton = screen.getByRole('button', { name: 'CSV İndir' });

      await user.click(exportButton);
      expect(mockDownload).toHaveBeenCalledWith(mockInsightsList);
    });
  });

  it('matches snapshot', () => {
    const { container } = renderInsightList({ insights: mockInsightsList });
    expect(container).toMatchSnapshot();
  });
});
