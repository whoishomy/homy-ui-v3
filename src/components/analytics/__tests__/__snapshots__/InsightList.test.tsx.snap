// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InsightList > matches snapshot 1`] = `
<div>
  <div>
    <div
      class="flex justify-between items-center mb-4"
    >
      <h2
        class="text-xl font-semibold text-gray-900"
      >
        İçgörüler
      </h2>
      <div
        class="flex items-center gap-4"
      >
        <select
          class="rounded-md border border-gray-300 text-sm"
        >
          <option
            value="all"
          >
            Tüm Kategoriler
          </option>
          <option
            value="PHYSICAL"
          >
            Fiziksel Aktivite
          </option>
          <option
            value="MENTAL"
          >
            Zihinsel Sağlık
          </option>
          <option
            value="NUTRITION"
          >
            Beslenme
          </option>
          <option
            value="SLEEP"
          >
            Uyku
          </option>
          <option
            value="undefined"
          >
            Sosyal Etkileşim
          </option>
          <option
            value="HEALTH"
          >
            Genel Sağlık
          </option>
        </select>
        <button
          class="justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 flex items-center gap-2"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-download w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 15V3"
            />
            <path
              d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
            />
            <path
              d="m7 10 5 5 5-5"
            />
          </svg>
          CSV İndir
        </button>
      </div>
    </div>
    <div
      class="space-y-4"
    >
      <article
        aria-label="Test message 1"
        class="relative bg-white rounded-lg shadow-lg p-4 mb-4"
        dir="ltr"
        role="article"
      >
        <div
          class="flex items-start justify-between"
        >
          <div
            class="flex items-center"
          >
            <span
              aria-hidden="true"
              class="text-xl p-2 rounded-full mr-3 text-green-500"
              data-testid="insight-icon-success"
            >
              ✓
            </span>
            <div>
              <p
                class="text-gray-900 font-medium"
              >
                Test message 1
              </p>
              <div
                class="mt-1 flex items-center text-xs text-gray-500"
              >
                <span>
                  Fiziksel Aktivite
                </span>
                <span
                  class="mx-2"
                >
                  •
                </span>
                <span>
                  25 Mayıs
                </span>
              </div>
            </div>
          </div>
          <button
            aria-label="Kapat"
            class="text-gray-400 hover:text-gray-600"
            tabindex="0"
            type="button"
          >
            ✕
          </button>
        </div>
        <div
          class="mt-3"
        >
          <h3
            class="text-xs font-medium text-gray-900 mb-1"
          >
            İlgili Metrikler
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Steps
            </span>
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Distance
            </span>
          </div>
        </div>
        <div
          class="mt-3"
        >
          <button
            aria-label="View details"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 w-full bg-blue-700 text-white hover:bg-blue-800"
            tabindex="0"
          >
            View details
          </button>
        </div>
      </article>
      <article
        aria-label="Test message 2"
        class="relative bg-white rounded-lg shadow-lg p-4 mb-4"
        dir="ltr"
        role="article"
      >
        <div
          class="flex items-start justify-between"
        >
          <div
            class="flex items-center"
          >
            <span
              aria-hidden="true"
              class="text-xl p-2 rounded-full mr-3 text-yellow-500"
              data-testid="insight-icon-warning"
            >
              ⚠
            </span>
            <div>
              <p
                class="text-gray-900 font-medium"
              >
                Test message 2
              </p>
              <div
                class="mt-1 flex items-center text-xs text-gray-500"
              >
                <span>
                  Beslenme
                </span>
                <span
                  class="mx-2"
                >
                  •
                </span>
                <span>
                  25 Mayıs
                </span>
              </div>
            </div>
          </div>
          <button
            aria-label="Kapat"
            class="text-gray-400 hover:text-gray-600"
            tabindex="0"
            type="button"
          >
            ✕
          </button>
        </div>
        <div
          class="mt-3"
        >
          <h3
            class="text-xs font-medium text-gray-900 mb-1"
          >
            İlgili Metrikler
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Steps
            </span>
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Distance
            </span>
          </div>
        </div>
        <div
          class="mt-3"
        >
          <button
            aria-label="View details"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 w-full bg-blue-700 text-white hover:bg-blue-800"
            tabindex="0"
          >
            View details
          </button>
        </div>
      </article>
      <article
        aria-label="Test message 3"
        class="relative bg-white rounded-lg shadow-lg p-4 mb-4"
        dir="ltr"
        role="article"
      >
        <div
          class="flex items-start justify-between"
        >
          <div
            class="flex items-center"
          >
            <span
              aria-hidden="true"
              class="text-xl p-2 rounded-full mr-3 text-red-500"
              data-testid="insight-icon-error"
            >
              ⚠
            </span>
            <div>
              <p
                class="text-gray-900 font-medium"
              >
                Test message 3
              </p>
              <div
                class="mt-1 flex items-center text-xs text-gray-500"
              >
                <span>
                  Uyku
                </span>
                <span
                  class="mx-2"
                >
                  •
                </span>
                <span>
                  25 Mayıs
                </span>
              </div>
            </div>
          </div>
          <button
            aria-label="Kapat"
            class="text-gray-400 hover:text-gray-600"
            tabindex="0"
            type="button"
          >
            ✕
          </button>
        </div>
        <div
          class="mt-3"
        >
          <h3
            class="text-xs font-medium text-gray-900 mb-1"
          >
            İlgili Metrikler
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Steps
            </span>
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Distance
            </span>
          </div>
        </div>
        <div
          class="mt-3"
        >
          <button
            aria-label="View details"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 w-full bg-blue-700 text-white hover:bg-blue-800"
            tabindex="0"
          >
            View details
          </button>
        </div>
      </article>
      <article
        aria-label="Test message 4"
        class="relative bg-white rounded-lg shadow-lg p-4 mb-4"
        dir="ltr"
        role="article"
      >
        <div
          class="flex items-start justify-between"
        >
          <div
            class="flex items-center"
          >
            <span
              aria-hidden="true"
              class="text-xl p-2 rounded-full mr-3 text-green-500"
              data-testid="insight-icon-success"
            >
              ✓
            </span>
            <div>
              <p
                class="text-gray-900 font-medium"
              >
                Test message 4
              </p>
              <div
                class="mt-1 flex items-center text-xs text-gray-500"
              >
                <span>
                  Zihinsel Sağlık
                </span>
                <span
                  class="mx-2"
                >
                  •
                </span>
                <span>
                  25 Mayıs
                </span>
              </div>
            </div>
          </div>
          <button
            aria-label="Kapat"
            class="text-gray-400 hover:text-gray-600"
            tabindex="0"
            type="button"
          >
            ✕
          </button>
        </div>
        <div
          class="mt-3"
        >
          <h3
            class="text-xs font-medium text-gray-900 mb-1"
          >
            İlgili Metrikler
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Steps
            </span>
            <span
              class="inline-flex items-center px-2 py-1 rounded-full text-xs font-medium bg-gray-200 text-gray-900"
            >
              Distance
            </span>
          </div>
        </div>
        <div
          class="mt-3"
        >
          <button
            aria-label="View details"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 w-full bg-blue-700 text-white hover:bg-blue-800"
            tabindex="0"
          >
            View details
          </button>
        </div>
      </article>
    </div>
  </div>
</div>
`;
